### ISSUE TITLE
MIF denoising of target detection results

### ISSUE THREAD

**Issue by jpolchlo** [azavea/nasa-hyperspectral#131]
Multidimensional iterated filtering was introduced in #116.  Though the intended use for decorrelation did not pan out as hoped, there is an additional application for the MIF technique raised in [1]: as a postprocessing noise reduction scheme, to be applied to results from a target detection method such as a matched filter.  This consists of determining the first intrinsic mode function (IMF)—which generally captures the noise present in a signal—and subtracting it from the target detection result.

This procedure seems promising enough to attempt, and should run relatively quickly, as it only involves a single IMF computation.

#### References

[1] Cicone, A., Liu, J., & Zhou, H. (2016). [Hyperspectral chemical plume detection algorithms based on multidimensional iterative filtering decomposition](https://arxiv.org/pdf/1512.01979.pdf). Philosophical Transactions of the Royal Society A: Mathematical, Physical and Engineering Sciences, 374(2065), 20150196.

**Comment by jamesmcclain** [comment]
@jpolchlo If this work is still in progress, feel free to send me a link to a branch or WIP PR -- I can still provide some discussion.

**Comment by jpolchlo** [comment]
I attempted the MIF denoising.  Sample results:
![image](https://user-images.githubusercontent.com/18410961/128717894-fb1ca2f4-d044-4473-9392-c04e87bad6e7.png)
The left image is the original, the middle is the first IMF, the right is the denoised image.  

My assessment is that this is not worth doing, particularly for how long this result took to generate _with GPU acceleration_.  It's almost assured that there is a faster way to achieve similar results.

I've posted the [Jupyter notebook](https://gist.github.com/jpolchlo/07e58f5c6e05e3469c00d28750cd7108) as a gist for posterity, but I won't be revisiting this technique.  The linked paper above was a bust.
